ts41/get.d.ts

```diff
@@ -0,0 +1,82 @@
+import {Split} from '../source/split';
+import {Integers} from '../source/utilities';
+
+/**
+ * Gets a deeply-nested property from an object, like lodash's `get` method.
+ *
+ * Use-case: retrieve a property from deep inside an API response or other complex object.
+ *
+ * @example
+ * import { Get } from 'type-fest'
+ *
+ * interface ApiResponse {
+ *   hits: {
+ *     hits: Array<{
+ *       _id: string
+ *       _source: {
+ *         name: Array<{
+ *           given: string[]
+ *           family: string
+ *         }>
+ *         birthDate: string
+ *       }
+ *     }>
+ *   }
+ * }
+ *
+ * type Name = Get<ApiResponse, 'hits.hits[0]._source.name'> // Array<{ given: string[]; family: string }>
+ *
+ * @explanation
+ *
+ * This works by first splitting the path based on `.` and `[...]` characters into a tuple of string keys.
+ * Then it recursively uses the head key to get the next property of the current object, until there are no keys
+ * left. Number keys extract the item type from arrays, or are converted to strings to extract types from tuples
+ * and dictionaries with number keys.
+ */
+export type Get<Object, Path extends string> = GetWithPath<Object, ToPath<Path>>;
+
+/**
+ * Like @see Get but receives an array of strings as a path parameter.
+ */
+type GetWithPath<T, Keys extends string[]> =
```

Comment:
```suggestion
type GetWithPath<T, Keys extends readonly string[]> =
```

Should this be `readonly` so it also works with readonly arrays?

Reply:
Regardless, can you add some tests that uses readonly arrays? Just to ensure it works.

Reply:
Have made this readonly but this type isn't exported so can't be directly tested. That's fine though IMO - it also can't be incorrectly used since it's not exported.